src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ConstructorInitializerSymbolReferenceFinder.cs

```diff
@@ -76,22 +79,78 @@ protected override bool CanFind(IMethodSymbol symbol)
                 {
                     return true;
                 }
+                else if (methodSymbol.Parameters.IsDefaultOrEmpty
+                    && syntaxFactsService.IsConstructorToken(t))
+                {
+                    var containingType = semanticModel.GetEnclosingNamedType(t.SpanStart, cancellationToken);
+                    return containingType != null
+                        && containingType.BaseType != null
+                        && containingType.BaseType.Name == typeName;
```

Comment:
It's not sufficent to just to a name comparison.  We shoudl actually be comparing the type symbols.